kuwala/core/backend/app/controller/data_source/snowflake.py

```diff
@@ -0,0 +1,176 @@
+
+        for i in range(len(schemas)):
+            query = f"SHOW TABLES in {schemas[i]['schema']}"
+            cursor.execute(query)
+            for table in cursor.fetchall():
+                tables.append(table[1])
+            
+            if tables:
+                schemas[i]["categories"].append(dict(category='tables', tables=tables))
+                tables = []
+
+            query = f"SHOW VIEWS in {schemas[i]['schema']}"
+            cursor.execute(query)
+
+            for view in cursor.fetchall():
+                views.append(view[1])
+
+            if views:
+                schemas[i]['categories'].append(dict(category='views', tables = views ))
+                views = []
+        
+    finally:
+        cursor.close()
+
+    connection.close()
+
+    return schemas
+
+
+def get_columns(
+    connection_parameters: ConnectionParameters, schema_name: str, table_name: str
+):
+    columns_query = f"""
+            SELECT column_name, data_type
+            FROM information_schema.columns
+            WHERE table_schema = '{schema_name.upper()}' AND table_name = '{table_name.upper()}'
+    """
+    columns = send_query(
+        connection_parameters=connection_parameters, query=columns_query
+    )[1:]
+    columns = list(map(lambda c: dict(column=c[0], type=c[1].upper()), columns))
+
+    return columns
+
+
+def get_table_preview(
+    connection_parameters: ConnectionParameters,
+    schema_name: str,
+    table_name: str,
+    columns: list[str],
+    limit_columns: int,
+    limit_rows: int,
+) -> dict:
+    if not schema_name:
+        raise HTTPException(
+            status_code=400, detail="Missing query parameter: 'schema_name'"
+        )
+
+    if not limit_columns:
+        limit_columns = 200
+
+    if not limit_rows:
+        limit_rows = 300
+
+    if not columns:
+        columns_query = f"""
+            SELECT *
+            FROM {schema_name}.{table_name}
+            LIMIT 0
+        """
+        columns = send_query(
+            connection_parameters=connection_parameters, query=columns_query
+        )
+
+        columns_string = functools.reduce(
+            lambda c1, c2: f"{c1}, {c2}", columns[0][0:limit_columns]
+        )
+    else:
+        columns_string = functools.reduce(
+            lambda c1, c2: f"{c1}, {c2}", columns[0:limit_columns]
+        )
+
+    rows_query = f"""
+        SELECT {columns_string}
+        FROM {schema_name}.{table_name}
+        LIMIT {limit_rows}
+    """
+    rows = send_query(connection_parameters=connection_parameters, query=rows_query)
+    columns = rows.pop(0)
+
+    return dict(columns=columns, rows=rows)
+
+
+def update_dbt_connection_parameters(
+    profile_yaml: dict, connection_parameters: ConnectionParameters
+) -> dict:
+    dev_profile = profile_yaml["kuwala"]["outputs"]["dev"]
+    dev_profile["user"] = connection_parameters.user
+    dev_profile["pass"] = connection_parameters.password
```

Comment:
This needs to be `dev_profile["password"]` (see last review).

Reply:
Noted